src/effects/VST3/VST3Utils.cpp

```diff
@@ -238,3 +245,132 @@ bool VST3Utils::ParseAutomationParameterKey(const wxString& key, Steinberg::Vst:
    return false;
 
 }
+
+
+const std::vector<VST3Wrapper::ParameterInfo>* VST3Wrapper::getParameterInfos() const
+{
+   if (mParameterInfos.size() == 0)
+   {
+      if (mEditController == nullptr)
+         return nullptr;
+
+      for (int i = 0, count = mEditController->getParameterCount(); i < count; ++i)
+      {
+         Steinberg::Vst::ParameterInfo vstParameterInfo{ };
+         if (mEditController->getParameterInfo(i, vstParameterInfo) == Steinberg::kResultOk)
+         {
+            // Add all parameterinfos, regardless of whether they can be automated or not,
+            // because different flags of them will be checked by different methods.
+            mParameterInfos.push_back(vstParameterInfo);
+         }
+      }
+   }
+
+   return &mParameterInfos;
+}
+
+
+bool VST3Wrapper::ForEachParameter(ParameterVisitor visitor) const
+{
+   const auto pParameterInfos = getParameterInfos();
+   if (pParameterInfos == nullptr)
+      return false;
+ 
+   for (const auto& parameterInfo : *pParameterInfos)
+   {
+      if ( ! visitor(parameterInfo) )
+         break;
+   }
+
+   return true;
+}
+
+
+bool VST3Wrapper::AtLeastOne(ParameterVisitor visitor) const
```

Comment:
Maybe you can call this `AnyOf` by analogy with `std::any_of`

Reply:
ok, done